1-JavaScript/Week3/homework/ex1-doubleEvenNumbers.test.js

```diff
@@ -11,14 +11,14 @@ Let's rewrite it (or _refactor_ it, as experienced developers would call it):
 ------------------------------------------------------------------------------*/
 // ! Function to be tested
 function doubleEvenNumbers(numbers) {
-  // TODO rewrite the function body using `map` and `filter`.
-  const newNumbers = [];
-  for (let i = 0; i < numbers.length; i++) {
-    if (numbers[i] % 2 === 0) {
-      newNumbers.push(numbers[i] * 2);
-    }
-  }
-  return newNumbers;
+  const evenNumbers = numbers.filter((even) => {
+    return even % 2 === 0;
+  });
+
+  const doubleNumbers = evenNumbers.map((num) => {
+    return num * 2;
+  });
+  return doubleNumbers;
```

Comment:
Awesome, good work! 💪 You can make this a bit simpler though, by chaining the `filter` and `map` methods together. See if you can find out how 🙂 I provided a hint in the code suggestion below:

```suggestion
  // Chain .filter and .map together, so first use filter and then map
  // Then you don't need two separate variables and you can just return
  // the result from filtering and mapping directly

  const evenNumbers = numbers.filter((even) => {
    return even % 2 === 0;
  });

  const doubleNumbers = evenNumbers.map((num) => {
    return num * 2;
  });
  return doubleNumbers;
```